include/envoy/event/dispatcher.h

```diff
@@ -86,13 +86,16 @@ class Dispatcher {
    * (http://www.haproxy.org/download/1.5/doc/proxy-protocol.txt)
    * @param use_orig_dst if a connection was redirected to this port using iptables,
    *        allow the listener to hand it off to the listener associated to the original port
+   * @param per_connection_buffer_limit_bytes soft limit on size of the listener's new connection
+   *        read and write buffers.
    * @return Network::ListenerPtr a new listener that is owned by the caller.
    */
   virtual Network::ListenerPtr createListener(Network::ConnectionHandler& conn_handler,
                                               Network::ListenSocket& socket,
                                               Network::ListenerCallbacks& cb,
                                               Stats::Store& stats_store, bool bind_to_port,
-                                              bool use_proxy_proto, bool use_orig_dst) PURE;
+                                              bool use_proxy_proto, bool use_orig_dst,
```

Comment:
Sorry, to ask you to do this, but I have a feeling we are going to keep adding params/options here. (I know soon we will split use_original_dst into use_orginal_port/use_original_address, etc.). Can we define a ```struct ListenerOptions``` and just pass that. That way we don't have to change a gazillion mocks and callsites the next time we add a param.